data_analysis/requirements.txt

```diff
@@ -1,11 +1,18 @@
+-r ../scrape_data/requirements.txt
 boto3==1.21.21 # The version can also be removed to resolve conflict.
 pandas==1.4.3
 geopandas==0.11.1
 s3fs==2022.7.1
+s3path==0.5.2
 shapely==1.8.4
 jupyter==1.0.0
 typer[all]==0.6.1
 plotly==5.11.0
 chart-studio==1.1.0
 statsmodels==0.13.5
 weightedstats==0.4.1
+tqdm==4.64.0
```

Comment:
Thanks for this and sorry that original `requirements.txt` didn't work. Can you confirm that the original file works by changing line 10 in requirements.txt to PyQt5>=5.15.7 , line 5 of data_analysis/requirements.txt to shapely>=1.8.4 , and your Python version to 3.11.8?  I've managed to get this running on a virtual M1 running macOS 14. I used a virtual environment, although I'm not sure whether that makes a difference.

Reply:
Thanks for these suggestions. I needed to make one additional change to the s3path version because s3path 0.3.4 is incompatible with python 3.11.8 due to pathlib API changes: https://bugs.python.org/issue43012. But with that it otherwise works.

I do recommend updating requests
(https://github.com/chihacknight/chn-ghost-buses/pull/76/commits/2b0c261e46ebf771261a20c2fd0fc1f180f18c1c) but that can be split into a separate PR.

Reply:
That's fine, you can update `requests` here if you'd like. Thanks for catching the vulnerability issue.